### ISSUE TITLE
Test Keycloak integration via Docker

### ISSUE THREAD

**Issue by kenfinnigan** [thorntail/thorntail#934]

**Comment by sberyozkin** [comment]
Hi Ken, I can start having a look next week or so

**Comment by sberyozkin** [comment]
Hi Ken, and Bob, I've had some initial look at it, what is the plan about configuring the login-conf and security constraints ? Should we something similar to v2, having the deployment specific KC adapter config ? thanks

**Comment by kenfinnigan** [comment]
I think that's already part of the keycloak module in v4, unless I'm mistaken.

**Comment by sberyozkin** [comment]
Not that I can see at the moment. I'll experiment...

**Comment by bobmcwhirter** [comment]
Should leverage login conf annotations from MP

**Comment by sberyozkin** [comment]
Bob, do you mean org.eclipse.microprofile.auth.LoginConfig ? Yes, seeing LoginConfigDeploymentsCustomizer...That eclipse package has no security constraints annotation though at the moment.
It might make sense to support both the annotation-driven and properties-driven config...

**Comment by bobmcwhirter** [comment]
The security constraints are then applied via normal JavaEE annotations. Or
I might be misunderstanding.

**Comment by sberyozkin** [comment]
Yes, I just stopped short of opening a new MP-JWT issue :-). I'm not sure KC deals with @RolesAllowed attached to the JAX-RS resources, and instead reacts to the security-constraints (web.xml). MP-JWT itself probably can, but not sure about KC.

**Comment by bobmcwhirter** [comment]
I think it does along with the whole emptyRolesSemantic stuff. Worth
verifying though. All web.xml stuff can be replaced with annotations if you
include MP LoginConfig. In theory.
